src/pudl/metadata/classes.py

```diff
@@ -1632,3 +1633,45 @@ def to_sql(
                 check_values=check_values,
             )
         return metadata
+
+
+class CodeData(Base):
+    """
+    A collection of dictionaries representing codes for various PUDL fields.
+
+    Used to export to documentation.
+    """
+
+    code_list: List[dict]
+
+    @classmethod
+    def from_code_names(
+        cls, code_names: Iterable[str], csv_dir: String
```

Comment:
Since we're passing in a `pathlib.Path` for `csv_dir` does it make sense to just keep it as a Path rather than casting to a String and needing to turn it back into some kind of Path?  It looks like Pydantic [has built-in support](https://pydantic-docs.helpmanual.io/usage/types/#pydantic-types) for Path, FilePath, and DirectoryPath types.